swiftsimio/reader.py

```diff
@@ -1391,44 +1497,55 @@ def generate_dataset(particle_metadata: SWIFTParticleTypeMetadata, mask):
     Here we loop through all of the possible properties in the metadata file.
     We then use the builtin property() function and some generators to
     create setters and getters for those properties. This will allow them
-    to be accessed from outside by using SWIFTParticleDataset.name, where
+    to be accessed from outside by using SWIFTGroupDatasets.name, where
     the name is, for example, coordinates.
 
     Parameters
     ----------
-    particle_metadata : SWIFTParticleTypeMetadata
-        the metadata for the particle type
+    group_metadata : SWIFTGroupMetadata
+        the metadata for the group
     mask : SWIFTMask
-        the mask object for the dataset
+        the mask object for the datasets
     """
 
-    filename = particle_metadata.filename
-    particle_type = particle_metadata.particle_type
-    particle_name = particle_metadata.particle_name
-    particle_nice_name = metadata.particle_types.particle_name_class[particle_type]
+    filename = group_metadata.filename
+    group = group_metadata.group
+    group_name = group_metadata.group_name
+
+    # Set nice name for group
+    if group_metadata.metadata.filetype == "snapshot":
+        group_nice_name = metadata.particle_types.particle_name_class[group]
+    elif group_metadata.metadata.filetype == "FOF":
+        group_nice_name = "FOFGroups"
+    elif group_metadata.metadata.filetype == "SOAP":
+        group_nice_name = metadata.soap_types.get_soap_name_nice(group)
```

Comment:
Why if we wrote all this code are we doing this three different ways for three catalogues that we wrote..?